src/runners/alert_handler.py

```diff
@@ -57,54 +57,88 @@ def log_failure(ctx, alert, e):
+    ctx.cursor().execute(query)
+
+# We get a list of alerts that don't have tickets. For each alert, check the group_id of the alert; if there is no alert that has that group_id and a ticket_id, create a ticket for that alert.
+# if there is an alert with a matching group_id and a ticket_id, update the body of the ticket with the new alert instead.
 
 
 def main():
     ctx = db.connect_and_execute(f'USE DATABASE {DATABASE};')
     alerts = get_new_alerts(ctx)
-    print('Found', len(alerts), 'new alerts to handle.')
+    log.info('Found', len(alerts), 'new alerts to handle.')
 
     for row in alerts:
         try:
             alert = json.loads(row[0])
+            group_id = row[7]
         except Exception as e:
             log.error("Failed unexpectedly", e)
             continue
-        print('Creating ticket for alert', alert)
-
-        # Create a new ticket in JIRA for the alert
-        try:
-            ticket_id = create_jira.create_jira_ticket(
-                alert_id=alert['ALERT_ID'],
-                query_id=alert['QUERY_ID'],
-                query_name=alert['QUERY_NAME'],
-                environment=alert['ENVIRONMENT'],
-                sources=alert['SOURCES'],
-                actor=alert['ACTOR'],
-                object=alert['OBJECT'],
-                action=alert['ACTION'],
-                title=alert['TITLE'],
-                event_time=alert['EVENT_TIME'],
-                alert_time=alert['ALERT_TIME'],
-                description=alert['DESCRIPTION'],
-                detector=alert['DETECTOR'],
-                event_data=alert['EVENT_DATA'],
-                severity=alert['SEVERITY'])
-        except Exception as e:
-            log.error(e, f"Failed to create ticket for alert {alert}")
-            log_failure(ctx, alert, e)
-            continue
+        log.info('Creating ticket for alert', alert)
+
+        # We check against the group ID for alerts in that group with the same ticket
+        query = f"""SELECT * from {ALERTS_TABLE} where GROUP_ID = '{group_id}' and TICKET is not null
+                    ORDER BY EVENT_TIME DESC
+                    LIMIT 1
+                """
+        if group_id is None:
+            correlated_results = []
+        else:
+            correlated_results = ctx.cursor().execute(query).fetchall()
+
+        log.info(f"Discovered {len(correlated_results)} correlated results")
+
+        if len(correlated_results) > 0:
+
+            # There is a group with a ticket that exists, so we should append to that ticket
+            ticket_id = correlated_results[0][3]
+            log.info(f"Found ticket {ticket_id}")
+            ticket_status = create_jira.check_ticket_status(ticket_id)
+            log.info(f"The status of the ticket is: {ticket_status}")
+
+            if str(ticket_status) == 'To Do':  # the status of the jira ticket doesn't get returned as a string; it gets returned as a class and a straight comparison to a string will always fail
```

Comment:
I figured it would be more useful down the line to have a method which gets us the status, and then if we need information about the status for some other feature we already have it; what's the reasoning for making the function more restrictive?

I believe there's just the one library, unfortunately.